online-shop/src/Components/EditProductPage.tsx

```diff
@@ -0,0 +1,262 @@
+                            </div>
+                        </div>
+                    </div>
+                );
+            } else if (this.props.submitChangesResponse === 204) {
+                return (
+                    <div className="container has-text-centered">
+                        <h1 className="has-text-success has-text-weight-semi-bold is-size-3">Successfully updated product!</h1>
+                        <Link to="/products">
+                            <button className="button is-text has-text-success is-medium">
+                                Back to products catalog
+                            </button>
+                        </Link>
+                    </div>
+                );
+            } else if (this.props.submitChangesResponse === 404) {
+                return (
+                    <div className="container has-text-centered">
+                        <h1 className="has-text-danger has-text-weight-semi-bold is-size-3">Ops, could not perform operation. Product not found!</h1>
+                        <Link to="/products">
+                            <button className="button is-text has-text-danger is-medium">
+                                Back to products catalog
+                            </button>
+                        </Link>
+                    </div>
+                );
+            } else if (this.props.submitChangesResponse === 999) {
+                return (
+                    <div className="container has-text-centered">
+                        <h1 className="has-text-danger has-text-weight-semi-bold is-size-3">There has been a problem while communicating with the server, please check back later.</h1>
+                        <Link to="/products">
+                            <button className="button is-text has-text-danger is-medium">
+                                Back to products catalog
+                            </button>
+                        </Link>
+                    </div>
+                );
+            }
+        }
+    }
+}
+
+const mapStateToProps = (state: AppState, additionalState: AdditionalEditProductPageState) => ({
+    match: additionalState.match,
+    productInEditStage: state.editProductReducer.productInEditStage,
+    isLoading: state.editProductReducer.isLoading,
+    submitChangesResponse: state.editProductReducer.submitChangesResponse
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+    loadProductData: (productID: number) => dispatch({ type: LOAD_PRODUCT_DETAILS_FOR_EDIT, payload: productID }),
+    editProduct: (product: IProduct) => dispatch({type : EDIT_PRODUCT, payload : product}),
+    initialProductDetailsLoad: () => dispatch(initialProductDetailsLoad())
+});
+
+const EditProductPageInitializer = connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(EditProductPage);
+
+export default EditProductPageInitializer;
```

Comment:
This is actually a Smart component (or a Container component - depends what naming convention you want to use). Please move it to a different file.